lib/Bio/EnsEMBL/DataCheck/Checks/ControlledTablesCompara.pm

```diff
@@ -0,0 +1,174 @@
+=head1 LICENSE
+
+Copyright [2018-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::ControlledTablesCompara;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::DataCheck::Utils qw/ hash_diff /;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'ControlledTablesCompara',
+  DESCRIPTION    => 'Controlled tables are consistent with compara master database',
+  GROUPS         => ['controlled_tables', 'compara', 'compara_master'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['genome_db', 'mapping_session', 'method_link', 'method_link_species_set', 'ncbi_taxa_node', 'species_set']
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $helper = $self->dba->dbc->sql_helper;
+
+  if ($self->dba->dbc->dbname !~ /master/) {
+    my $master_tables = [
+      'genome_db',
+      'mapping_session',
+      'method_link',
+      'method_link_species_set',
+      'species_set'
+    ];
+    $self->master_tables($helper, $master_tables);
+  }
+
+  # The ncbi_taxa_name table has rows added in the compara db;
+  # so don't check that, ncbi_taxa_node is sufficient to check
+  # whether we're up to date.
+  my $taxonomy_tables = ['ncbi_taxa_node'];
+  $self->taxonomy_tables($helper, $taxonomy_tables);
+}
+
+sub master_tables {
+  my ($self, $helper, $tables) = @_;
+
+  # This test requires a compara_master database in the registry,
+  # where the 'species' parameter is the same as the database name.
+  my $desc_1 = "Compara master database found";
+  my $master_db_name = 'ensembl_compara_master';
+  my $division = $self->dba->get_division;
+  if ($division ne 'vertebrates') {
+    $master_db_name =~ s/ensembl/$division/;
+  }
```

Comment:
In Compara we have a different naming scheme. We name our master databases `ensembl_compara_master(_$DIVISION)`, but they are always named `compara_master` in our Registry config files to simplify cross-divisions work.

I could make this DC try both schemes ? Otherwise, we could add a command-line argument in `run_datachecks.pl` that holds the URI of the master database (like `old_server_uri`).

Reply:
If the convention is to name it `compara_master` we can just go with that, and try to keep things simple. As long as the non-vert teams are aware of it, they can provide an appropriate registry.